### PR TITLE
Fix sec vuln with list of claims

### PR DESCRIPTION
**Pull request by Waterdrips**
This PR adds a fix for https://github.com/dgrijalva/jwt-go/issues/422

Tests were added for failing and passing states then the code was updated for the case in the JWT Spec that allows a list of "aud" as well as a single string "aud"

Signed-off-by: Alistair Hey <alistair@heyal.co.uk>

### CODE REVIEW

File: example_test.go [hunk 1]
```diff
@@ -2,7 +2,7 @@ package jwt_test
 
 import (
 	"fmt"
-	"github.com/dgrijalva/jwt-go"
+	"github.com/form3tech-oss/jwt-go"
```
Comment (sev3ryn) [inline example_test.go:5]: Why are this changes needed in example_test.go?
Functions should be named as Example - to have examples in godoc.
Also some external jwt-go is imported

File: map_claims.go [hunk 2]
```diff
@@ -13,7 +13,15 @@ type MapClaims map[string]interface{}
 // Compares the aud claim against cmp.
 // If required is false, this method will return true if the value matches or is unset
 func (m MapClaims) VerifyAudience(cmp string, req bool) bool {
-	aud, _ := m["aud"].(string)
+	aud, ok := m["aud"].([]string)
+	if !ok {
+		strAud, ok := m["aud"].(string)
+		aud = append(aud, strAud)
+		if !ok {
```
Comment (sev3ryn) [inline map_claims.go:20]: nit: `if` should be before `aud = append(aud, strAud)`

**Comment by Waterdrips** [comment]
oh sorry - we decided to fork and fix in our company org and I left this PR open

**Comment by sev3ryn** [comment]
but why closing it? :) 

I would be great to have vulnerability fix merged into upstream

**Comment by camin-mccluskey** [comment]
@Waterdrips Echoing @sev3ryn's comment - it would be ideal to have this fix live in the latest version. This is now categorised as a high severity vulnerability https://snyk.io/vuln/golang:github.com%2Fdgrijalva%2Fjwt-go and means we can no longer use the library in an enterprise environment

**Comment by Waterdrips** [comment]
> but why closing it? :)
> 
> I would be great to have vulnerability fix merged into upstream

theres an open PR addressing this from Match #385 which has not been addressed. 

We have taken the decision to fix this on our oss fork rather than wait for some activity here.

**Comment by alexellis** [comment]
@dgrijalva if you have time to merge a fix like this, then we can all dump our forks to /dev/null. What are your thoughts?

**Comment by aleksandrzhiliaev** [comment]
Are we going to fix and bump the new version of the library?
It is still an issue and a lot of projects rely on it:
https://www.whitesourcesoftware.com/vulnerability-database/CVE-2020-26160

**Comment by brianmay** [comment]
See #286

**Pull request closed without merging**